src/adapter/src/rbac.rs

```diff
@@ -156,21 +156,30 @@ pub fn check_plan(
         return Ok(());
     }
 
-    if session.is_superuser() {
-        return Ok(());
-    }
-
     let role_id = session.role_id();
-    let Some(role) = catalog.try_get_role(role_id) else {
+    if catalog.try_get_role(role_id).is_none() {
         // PostgreSQL allows users that have their role dropped to perform some actions,
         // such as `SET ROLE` and certain `SELECT` queries. We haven't implemented
         // `SET ROLE` and feel it's safer to force to user to re-authenticate if their
         // role is dropped.
         return Err(AdapterError::ConcurrentRoleDrop(role_id.clone()));
     };
 
+    if session.is_superuser() {
+        return Ok(());
+    }
+
+    let all_roles: Vec<_> = catalog
+        .collect_role_membership(role_id)
+        .into_iter()
+        .map(|id| catalog.get_role(&id))
+        .collect();
```

Comment:
If you remove the collect and don't hard code this to a vec, I think things still work because the .any below only needs an Iter?

Reply:
Good catch, updated.